UIPS/UIP-0131.md

```diff
@@ -0,0 +1,121 @@
+
+In cases of rollbacks, %jael must ensure that its state synthesis algorithm appropriately handles the reversion without compromising the integrity of cross-chain point locations. This becomes particularly important when points are in transit between chains.
+
+TODO: Detail exact state synthesis algorithm for handling multi-chain state reconciliation.
+
+### Base Implementation
+
+Base's optimistic rollup architecture introduces specific timing requirements for the return path of the bridge protocol. While points can begin operating immediately upon arrival on Base, returns to Ethereum require waiting for Base's 7-day challenge period to expire. This ensures the Embassy contract state is final before the Ministry vault releases points back to Ethereum.
+
+Oracle validation windows adapt to Base's block confirmation schedule, requiring a minimum of 15 L2 blocks before accepting transfer tickets. The Embassy contract implements a local timeout mechanism to handle potential state synchronization issues between Base and Ethereum during return transfers.
+
+TODO: Specify exact block confirmation thresholds and timeout parameters.
+
+Rationale
+---------
+
+The design establishes Ethereum as the senior chain to preserve the existing security model while enabling point mobility. Using galaxies as oracles leverages the network's existing power structure and economic incentives rather than introducing external validators. Points maintain complete functionality across chains through comprehensive state transfer.
+
+Base's position as an Ethereum L2 makes it an ideal first implementation target. Its Ethereum compatibility simplifies transaction verification, while its established security model reduces implementation risks. The immediate benefit of lower transaction costs provides value to users while proving the architecture for future chain integrations.
+
+The ticket-based transfer mechanism creates explicit, verifiable state transitions with clear security checkpoints. This enables efficient validation while ensuring atomic transfers between chains, preventing any state where a point could exist simultaneously on multiple chains.
+
+Alternative approaches considered include trustless bridging (rejected due to complexity and security risks), multi-primary chains (which would complicate governance), lightweight state transfer (limiting functionality), and integration with official cross-chain bridges. While integration with established bridge providers merits further exploration, the current design optimizes for sovereignty and operational consistency while maintaining upgradeability.
```

Comment:
> optimizes for sovereignty

Can you elaborate? Increasing the amount of non-urbit-native places that urbit assets can be transferred to feels like it's in conflict with that.

Reply:
I'll post my recent USTJ article as the 'longer form' version of how I think about this: https://urbitsystems.tech/article/v01-i02/on-the-utility-of-azimuth-and-decentralized-identity

Shorter form is: https://sarlev-sarsen.rooftopdao.io/blog/why-l1masterrace-is-more-than-a-meme

TL;DR: You want your identity layer to be as high-level as possible. Putting your digital identifier on the most legible system, and additionally having it portable between global consensus systems increases your sovereignty by instantiating a right to exit. This is why the naive rollup is such an abomination in my eyes; it both decreases legibility, and totally nukes the right to exit.